vizro-core/src/vizro/models/types.py

```diff
@@ -248,6 +251,38 @@ def _check_type(cls, captured_callable: CapturedCallable, field: ModelField) ->
         return captured_callable
 
 
+@contextmanager
+def _pio_templates_default():
+    """Sets pio.templates.default to "vizro_dark" and then reverts it.
+
+    This is to ensure that in a Jupyter notebook captured charts look the same as when they're in the dashboard. When
```

Comment:
I think this is the one thing I don't fully grasp: are we now intending the Jupyter chart in Vizro style to look 100% (incl. margins) like the dashboard case, or not, ie all colors etc, but NOT the margins.

Reply:
Good question. At the moment the charts look 100% the same (including margins) and the template is optimised for charts in a dashboard.

After we complete "Encode different template logic for figures used inside / outside of dashboard" in https://github.com/McK-Internal/vizro-internal/issues/937 then the charts will look 99% the same since the margins would then be different for a chart used inside vs. outside a dashboard.